### PR TITLE
Decouple JsonElements from JsonDocument.

### PR DESCRIPTION
**Pull request by brentschmaltz**
In some multi threaded situations the incorrect JsonElement could have been returned.
Each JsonProperty is stored in a dictionary as a <name, JsonElement>.
First cloning the JsonElement to detach from the JsonDocument.

The JsonDocument is now correctly disposed by the creator.

### CODE REVIEW

File: src/Microsoft.IdentityModel.JsonWebTokens/Json/JsonClaimSet.cs [hunk 1]
```diff
@@ -25,8 +25,8 @@ internal class JsonClaimSet
 
         internal JsonClaimSet(JsonDocument jsonDocument)
         {
-            RootElement = jsonDocument.RootElement.Clone();
-            jsonDocument.Dispose();
```
Comment (keegan-caruso) [inline src/Microsoft.IdentityModel.JsonWebTokens/Json/JsonClaimSet.cs:29]: can you elaborate on why the Dispose is not needed anymore?

Comment (keegan-caruso) [reply src/Microsoft.IdentityModel.JsonWebTokens/Json/JsonClaimSet.cs:29]: oh I see, they get disposed where they are created now, resolving

File: test/Microsoft.IdentityModel.JsonWebTokens.Tests/JsonWebTokenTests.cs [hunk 2]
```diff
@@ -43,6 +44,117 @@ public class JsonWebTokenTests
             new Claim("dateTimeIso8061", dateTime.ToUniversalTime().ToString("o", CultureInfo.InvariantCulture), ClaimValueTypes.DateTime, "LOCAL AUTHORITY", "LOCAL AUTHORITY"),
         };
 
+#if !NET452
```
Comment (jennyf19) [inline test/Microsoft.IdentityModel.JsonWebTokens.Tests/JsonWebTokenTests.cs:47]: NET45? not NET452?

**Review by jennyf19 — APPROVED** [review]

**Review by keegan-caruso — APPROVED** [review]

File: test/Microsoft.IdentityModel.JsonWebTokens.Tests/JsonWebTokenTests.cs (hunk 2 above)
Comment (brentschmaltz) [reply test/Microsoft.IdentityModel.JsonWebTokens.Tests/JsonWebTokenTests.cs:47]: @jennyf19 the test environment uses 452 as xunit is not available for 451.

**Pull request merged**